tools/mkldscript.c

```diff
@@ -193,13 +208,55 @@ static void write_ld_script(FILE *fout)
         //fprintf(fout, "\n    }\n");
     }
 
-
-    fputs("    _RomEnd = _RomSize;\n}\n", fout);
+    fputs("    _RomEnd = _RomSize;\n\n", fout);
+
+    // Debugging sections
```

Comment:
I'm rather surprised that all this has to be set up manually. Is that because we're using an explicit linker script?

Reply:
I changed the /DISCARD/ at the bottom to remove all sections that aren't explicitly mentioned. From what I understand this is better practice and is a straight improvement from just discarding a couple of known problematic sections when there could be more. Due to this though, I had to specify the debugging sections explicitly so they aren't also discarded.